### PR TITLE
MastClass query helper function

### PR DESCRIPTION
**Pull request by mstcyr2**
The MAST API requires users to use JSON/Python Dictionaries to query MAST services. We want to create a helper function to format parameter dictionaries to make `astroquery.mast.Mast` queries easier for the users.

### CODE REVIEW

File: docs/mast/mast_mastquery.rst
Comment (jaymedina) [inline docs/mast/mast_mastquery.rst:13]: I would reword this slightly to: _The basic MAST query function allows users to query through the following `MAST Services <https://mast.stsci.edu/api/v0/_services.html>`__ using their corresponding parameters_ (with the exception of the service that you mentioned `mast_query` doesnt account for).

File: docs/mast/mast_mastquery.rst
Comment (jaymedina) [inline docs/mast/mast_mastquery.rst:22]: Be sure to mention that the criteria under the `filters` parameter work just the same way as the criteria used in `Observations.query_criteria` for example. Would be good to draw parallels to `ObservationsClass` as its familiar to our users.

File: docs/mast/mast_mastquery.rst
Comment (jaymedina) [inline docs/mast/mast_mastquery.rst:30]: Where can people go to see what options they have for `filters` and `columns`? i.e. How would a user know that the column name is `instrume` and not `instrument`? Be sure to post a resource pointing to this.

File: docs/mast/mast_mastquery.rst
Comment (jaymedina) [inline docs/mast/mast_mastquery.rst:108]: Is this the service that ignores when the user specify which `columns` to return? If so, be sure to mention this somewhere in here before the example. Is there error handling to catch when this happens?

File: docs/mast/mast_mastquery.rst
Comment (jaymedina) [inline docs/mast/mast_mastquery.rst:155]: I'm a little confused about this one, why was columns changes to 'ra' from '*'? This example is meant to show how not using the right filters leads to a `TraceBackError`, so what about adding a non-coordinate filter to the cone search, since cone search expects RA/Dec and not Teff, for example?

File: docs/mast/mast_mastquery.rst
Comment (jaymedina) [reply docs/mast/mast_mastquery.rst:108]: Nvm! I see that you describe this a few examples below. You can ignore this comment

File: docs/mast/mast_mastquery.rst
Comment (jaymedina) [inline docs/mast/mast_mastquery.rst:180]: There's a typo here, the a before `~astropy.table.Table`

File: astroquery/mast/observations.py
Comment (jaymedina) [inline astroquery/mast/observations.py:918]: Currently it seems like the input `service` argument is case-sensitive, and I think people will have trouble with that if they don't capitalize the right letters. Can you make the `service` argument case-insensitive?

**Review by jaymedina — CHANGES_REQUESTED** [review]
Looking good so far, thanks for this work! Let me know if you have any questions about my requested changes and/or comments.

File: docs/mast/mast_mastquery.rst
Comment (mstcyr2) [reply docs/mast/mast_mastquery.rst:155]: Columns value is defaulted to '*' when not specified so I just changed it to something users might be looking for in a cone search. And I see what you mean about the other filters, I will add that.

**Review by jaymedina — APPROVED** [review]
This looks good to me! I will defer to @bsipocz for any other comments.

File: astroquery/mast/observations.py
Comment (bsipocz) [inline astroquery/mast/observations.py:911]: we're trying to get rid of the ``**kwargs`` in all signatures, at least where it's not a convenience method falling back on totally different signatures (think about `Table.read()`), and where the list of potential kwargs are not unmanagably long. 

So, if possible, please spell them out.

Comment (bsipocz) [reply astroquery/mast/observations.py:911]: Also, please make optional agruments kwarg only. And please add documentation as docstring

**Review by bsipocz — CHANGES_REQUESTED** [review]
I have two things before we can merge this:

- docstrings are needed for any public functionality
- please avoid using the `**kwargs` syntax in a signature as it masks over potential issues, keywords the user think have any effect while they are ignored, etc. We do this type of cleanup slowly all over the library, so no new things should add one, unless there is a very convincing argument along with it.

File: astroquery/mast/observations.py
Comment (mstcyr2) [reply astroquery/mast/observations.py:911]: I think `**kwargs` are necessary here because `mast_query` deals with requests to all the different mashup services and uses kwargs for each service's list of properties. I also addressed the mashup property parameters using kwargs because they're all optional for the `Mast.service_request` call in the return statement. The api also does error-handling for properties or kwargs that don't exist within a service.

Comment (jaymedina) [reply astroquery/mast/observations.py:911]: Hi @bsipocz , Makayla is right. The arguments here depend on which database is being accessed, so we need `**kwargs` as a catchall. What do you mean by adding documentation as docstring? In this PR Makayla added a link to the MAST documentation a few lines below (L924 currently).

**Comment by jaymedina** [comment]
Just a heads up, I'll be taking over this PR for Makayla since she completed her internship ( 🎉 )

**Comment by bsipocz** [comment]
Congrats @mstcyr2 on finishing your internship. And thank you for all the contributions to astroquery.

@jaymedina - could you have a look at the remote tests, e.g. I see failures mast_mastquery.rst.

**Comment by jaymedina** [comment]
> @jaymedina - could you have a look at the remote tests, e.g. I see failures mast_mastquery.rst.

Thanks for looking into this. Is it that you saw doc build failures in `mast_mastquery.rst`? I just built the docs locally with no errors and then ran each of the examples manually in a notebook and all seems to work fine. My latest commit fixes syntax in the examples by removing the spaces around `=` in the kwarg calls, which I suspect might have led to what you were seeing? I also just ran the `test_mast_remote.py` unit tests and there are no failures related to the MAST query helper function. Let me know if there is something else I should check. 

Thanks!

**Comment by bsipocz** [comment]
> Thanks for looking into this. Is it that you saw doc build failures caused by mast_mastquery.rst

No, I see failures when run the doctests in that file.

``pytest docs/mast -R``

**Comment by bsipocz** [comment]
This needs a rebase, I tried to use the "newsish" github conflict resolution with a rebase, but it still only was able to do the merge approach.

**Comment by jaymedina** [comment]
Just did a rebase and squash @bsipocz let me know if there's anything else I should update. Thanks!

**Review by bsipocz — APPROVED** [review]

**Pull request merged**
